Classes/Global State Explorers/FLEXFileBrowserTableViewController.m

```diff
@@ -265,4 +300,82 @@ - (void)openFileController:(NSString *)fullPath
     self.documentController = controller;
 }
 
+- (void)fileBrowserRename:(UITableViewCell *)sender
+{
+    NSString *fullPath = nil;
+
+    NSIndexPath *indexPath = [self.tableView indexPathForCell:sender];
+    if (indexPath) {
+        NSString *subpath = [self.childPaths objectAtIndex:indexPath.row];
+        fullPath = [self.path stringByAppendingPathComponent:subpath];
+    } else {
+        indexPath = [self.searchDisplayController.searchResultsTableView indexPathForCell:sender];
+        fullPath = [self.searchPaths objectAtIndex:indexPath.row];
+    }
+
+    self.fileOperationController = [[FLEXFileBrowserFileRenameOperationController alloc] initWithPath:fullPath];
+    self.fileOperationController.delegate = self;
+    [self.fileOperationController show];
+}
+
+- (void)fileBrowserDelete:(UITableViewCell *)sender
+{
+    NSString *fullPath = nil;
+
+    NSIndexPath *indexPath = [self.tableView indexPathForCell:sender];
+    if (indexPath) {
+        NSString *subpath = [self.childPaths objectAtIndex:indexPath.row];
+        fullPath = [self.path stringByAppendingPathComponent:subpath];
+    } else {
+        indexPath = [self.searchDisplayController.searchResultsTableView indexPathForCell:sender];
+        fullPath = [self.searchPaths objectAtIndex:indexPath.row];
+    }
+
+    self.fileOperationController = [[FLEXFileBrowserFileDeleteOperationController alloc] initWithPath:fullPath];
+    self.fileOperationController.delegate = self;
+    [self.fileOperationController show];
+}
+
+- (void)reloadDisplayedPaths {
+    if (self.searchDisplayController.isActive) {
+        [self reloadSearchPaths];
+        [self.searchDisplayController.searchResultsTableView reloadData];
+    } else {
+        [self reloadChildPaths];
+        [self.tableView reloadData];
+    }
+}
+
+- (void)reloadChildPaths {
+    self.childPaths = [[NSFileManager defaultManager] contentsOfDirectoryAtPath:self.path error:NULL];
+}
+
+- (void)reloadSearchPaths {
+    self.searchPaths = nil;
+    self.searchPathsSize = nil;
+
+    //clear pre search request and start a new one
+    [self.operationQueue cancelAllOperations];
+    FLEXFileBrowserSearchOperation *newOperation = [[FLEXFileBrowserSearchOperation alloc] initWithPath:self.path searchString:self.searchString];
+    newOperation.delegate = self;
+    [self.operationQueue addOperation:newOperation];
+}
+
+@end
+
+
+@implementation FLEXFileBrowserTableViewCell
```

Comment:
What do you think about implementing the actions in `-tableView:performAction:forRowAtIndexPath:` rather than adding the custom cell subclass and responder chain forwarding? It would also cut out the `indexPath` lookup since that is provided directly to the `UITableViewDelegate` method.